python/pyproject.toml

```
@@ -63,7 +66,7 @@ boto3 = { version = ">=1.17.106", optional = true }
 s3fs = { version = ">=2021.08.0,<2024.1.0", optional = true } # Upper bound set arbitrarily, to be reassessed in early 2024.
 adlfs = { version = ">=2021.07.0,<2024.1.0", optional = true } # Upper bound set arbitrarily, to be reassessed in early 2024.
 psycopg2-binary = { version = ">=2.9.6", optional = true }
-sqlalchemy = { version = "^2.0.18", optional = true }
+sqlalchemy = { version = "^2.0.18" }
```

SQL Alchemy should only be installed when people are using the SQLCatalog.
```suggestion
sqlalchemy = { version = "^2.0.18", optional = true }
```

---

fixed.